src/Language/PureScript/CoreImp/ToJSON.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE NoOverloadedStrings #-}
+-- |
+-- Dump the core imperative representation in JSON format for consumption
+-- by third-party code generators
+--
+module Language.PureScript.CoreImp.ToJSON
+  ( Language.PureScript.CoreImp.ToJSON.moduleToJSON
+  ) where
+
+
+import Prelude.Compat
+
+import Data.Aeson
+import qualified Data.Text as T
+import Data.Version (Version)
+
+import Language.PureScript.AST (SourceSpan)
+import Language.PureScript.CoreFn (Module)
+import Language.PureScript.CoreFn.ToJSON as CoreFnToJSON
+import Language.PureScript.CoreImp.AST
+import Language.PureScript.PSString
+
+
+
+
+moduleToJSON :: (Show a) => Version -> Module a -> [AST] -> Value
+moduleToJSON v m a = object [ T.pack "body"    .= map (astToJSON) a
+                            , T.pack "corefn"  .= CoreFnToJSON.moduleToJSON v m -- might be good to "keep a snapshot of the original FP logic of the module" depending on coreimp-processing use-case (others can ignore it)
+                            , T.pack "modraw"  .= toJSON (show m) -- only way to capture type/ctors info without rewriting corefn, good enough for those needing it badly enough, others can ignore it
```

Comment:
What does `modraw` mean? Also, let's not use `show` here.

In general, I'd like to keep this minimal. It's harder to remove features later.

Reply:
Yeah I'll do away with this shortly, accidentally pushed this last commit. Still a bit more in flux than I expected I'm afraid!